busybox: Update to 1.37.0, fix cron log level

hnyman opened the pull request:
Update busybox to 1.37.0 and also fix the cron log level handling in it (due to upstream changes)

Main change has been discussed and debugged by @hauke and me in #16526, so this is just a rebased/fixed cherry-pick of that PR commit from @rsalvaterra plus the cron fix from me.

Tested with MT6000, DL-WRX36 and RT3200.

Reference to #16526

rsalvaterra committed:
busybox: update to 1.37.0

Add a patch to fix non-x86 builds, refresh patches and update/fix the
configuration as described in 98b09ba250 (cited here for future reference).

Config refresh:

Refresh commands, run after busybox is first built once:

cd package/utils/busybox/config/
../convert_menuconfig.pl ../../../../build_dir/target-arm_cortex-a15+neon-vfpv4_musl_eabi/busybox-default/busybox-1.37.0
cd ..
./convert_defaults.pl ../../../build_dir/target-arm_cortex-a15+neon-vfpv4_musl_eabi/busybox-default/busybox-1.37.0/.config > Config-defaults.in

Manual edits needed after config refresh:

* Config-defaults.in: OpenWrt config symbol IPV6 logic applied to
  BUSYBOX_DEFAULT_FEATURE_IPV6

* Config-defaults.in: OpenWrt config TARGET_bcm53xx logic applied to
  BUSYBOX_DEFAULT_TRUNCATE (commit 547f1ec)

* Config-defaults.in: OpenWrt logic applied to
  BUSYBOX_DEFAULT_LOGIN_SESSION_AS_CHILD (commit dc92917)

* Config-defaults.in: correct the default ports that get reset
  BUSYBOX_DEFAULT_FEATURE_HTTPD_PORT_DEFAULT    80
  BUSYBOX_DEFAULT_FEATURE_TELNETD_PORT_DEFAULT  23

* config/editors/Config.in: Add USE_GLIBC dependency to
  BUSYBOX_CONFIG_FEATURE_VI_REGEX_SEARCH (commit f141090)

* config/shell/Config.in: change at "Options common to all shells" the conditional symbol
  SHELL_ASH --> BUSYBOX_CONFIG_SHELL_ASH
  (discussion in http://lists.openwrt.org/pipermail/openwrt-devel/2021-January/033140.html
  Apparently our script does not see the hidden option while
  prepending config options with "BUSYBOX_CONFIG_" which leads to a
  missed dependency when the options are later evaluated.)

* Edit a few Config.in files by adding quotes to sourced items in
  config/Config.in, config/networking/Config.in and config/util-linux/Config.in (commit 1da014fcca)

Signed-off-by: Rui Salvaterra <rsalvaterra@gmail.com>
[rebased with the change download line commit]
[corrected version in the refresh example in commit message]
Signed-off-by: Hannu Nyman <hannu.nyman@iki.fi>
Link: https://github.com/openwrt/openwrt/pull/17107
Signed-off-by: Hauke Mehrtens <hauke@hauke-m.de>

namiltd commented:
All cron calls are still in the system log. They disappear only after option cronloglevel '9'

hnyman commented:
The logging depends of a possible cronloglevel setting in /etc/config/system.

To my knowledge, there is no such setting by default, but  it may been been put there in ancient times (or by LuCI).

There is no code that sets it by default:
https://github.com/search?q=repo%3Aopenwrt%2Fopenwrt+cronloglevel&type=code

pesa1234 commented:
`BUSYBOX_CONFIG_FEATURE_SYSLOG [=y]`
Sorry is it still requested?

hnyman commented:
It is an automatic dependency of several wanted features.

hnyman commented:
@namiltd 
The problem that you observed, is likely coming from LuCI, not busybox.
By default, a virgin just-flashed system with wither the old or new busybox does not have the cronloglevel set up. Then the init script uses the default set there, which is 6.

But LuCI seems to be eager to set it into 8, if the value is missing from /etc/config/system.
It is enough to visit the system config page and set any (other) value there, and the cronloglevel may also get set.
I will fix that in LuCI.

hnyman committed:
busybox: Fix cron log level to match previous behavior

Change the cron log level in init script from 5 to 7
in order to match previous log behavior and to avoid
log spam every minute.

Busybox cron only uses levels 5,7,8, where 5 is debug and
7 and 8 are normal logging. Use 7 as default.

(Fun note: this is needed because upstream corrected an
ancient bug in their own log level handling and now 5 actually
means something.)

Signed-off-by: Hannu Nyman <hannu.nyman@iki.fi>
Link: https://github.com/openwrt/openwrt/pull/17107
Signed-off-by: Hauke Mehrtens <hauke@hauke-m.de>

hnyman commented:
I fixed LuCI with https://github.com/openwrt/luci/commit/1e7ce8de990e0bcb36b77bdb846b644aa1724568

And I changed the init script to also default to 7, as 6 is not used in busybox crond code. (Only levels 5, 7 and 8 are used, and 7 is normal logging, while 5 is debugging)

Neustradamus commented:
@hnyman: Good job!

hnyman commented:
You need to change cron log from debug logging to normal...
And you need to restart cron service or reboot the router.
